File: src/obstacle/ObstaclePoint.cpp
// Copyright (c) TrajoptLib contributors

#include "obstacle/ObstaclePoint.h"

#include <iostream>

namespace trajopt {

ObstaclePoint::ObstaclePoint(double x, double y) : x(x), y(y) {}

std::ostream& operator<<(std::ostream& stream, const ObstaclePoint& point) {
  return stream << "{\"x\": " << point.x << ", \"y\": " << point.y << "}";
}
}  // namespace trajopt

File: java/src/main/java/org/sleipnirgroup/trajoptlib/ObstaclePoint.java
// Copyright (c) TrajoptLib contributors

package org.sleipnirgroup.trajoptlib;

public final class ObstaclePoint {
    public final double x;
    public final double y;

    public ObstaclePoint(double x, double y) {
        this.x = x;
        this.y = y;
    }

    @Override
    public String toString() {
        return "{\"x\": " + x
            + ", \"y\": " + y
            + "}";
    }
}
